hawkular-wildfly-agent/pom.xml

```diff
@@ -113,6 +113,7 @@
     <dependency>
       <groupId>org.jboss.logging</groupId>
       <artifactId>jboss-logging</artifactId>
+      <version>3.1.4.GA-redhat-2</version>
```

Comment:
What happens to an agent that is compiled with logging 3.1.4.GA but run inside EAP7+ ? If that works also, then this fix would be OK (make sure you add a small XML comment in the pom to briefly explain why that hardcoded version is in there).

If that doesn't work (that is, if we can't compile with 3.1.4.GA and have it run in EAP7+) that is going to mean we can't do what @pilhuhn  suggests and have common code run on both (since logging is all throughout everything, it would be hard to fix this - unless we write some kind of our own logging proxy and put something in that compatibility utility class you wrote.).